File: backend/src/services/counts.ts
import { TrackedActor } from '../types';
import { ALL_ACTOR_CLASSES } from '../constants/actorClasses';

export function buildCounts(tracks: TrackedActor[]) {
  const counts = Object.fromEntries(ALL_ACTOR_CLASSES.map((className) => [className, 0])) as Record<string, number>;

  for (const track of tracks) {
    counts[track.className] = (counts[track.className] || 0) + 1;
  }

  return {
    peaton: counts.peaton,
    motocicleta: counts.motocicleta,
    automovil: counts.automovil,
    bus_transcaribe: counts.bus_transcaribe,
    ciclista: counts.ciclista,
    cebra: counts.cebra,
    full: counts
  };
}

File: backend_py/app/counts.py
from __future__ import annotations

from .actor_classes import ALL_ACTOR_CLASSES


def build_counts(tracks: list[dict]) -> dict:
    counts = {name: 0 for name in ALL_ACTOR_CLASSES}
    for track in tracks:
        key = track.get("className")
        if key in counts:
            counts[key] += 1

    return {
        "peaton": counts["peaton"],
        "motocicleta": counts["motocicleta"],
        "automovil": counts["automovil"],
        "bus_transcaribe": counts["bus_transcaribe"],
        "ciclista": counts["ciclista"],
        "full": counts,
    }
